2_process/src/process.r

```diff
@@ -0,0 +1,40 @@
+library(dplyr)
+library(stringr)
+library(whisker)
+
+process_data<-function(data_file='model_RMSEs.csv', processed_data_file='model_summary_results.csv', diagnostic_output_file = 'model_diagnostic_text.txt'){
+	data_file_path = file.path("1_fetch","out",data_file)
+	# Prepare the data for plotting
+	eval_data <- readr::read_csv(data_file_path, col_types = 'iccd') %>%
+	  filter(str_detect(exper_id, 'similar_[0-9]+')) %>%
+	  mutate(col = case_when(
+		model_type == 'pb' ~ '#1b9e77',
+		model_type == 'dl' ~'#d95f02',
+		model_type == 'pgdl' ~ '#7570b3'
+	  ), pch = case_when(
+		model_type == 'pb' ~ 21,
+		model_type == 'dl' ~ 22,
+		model_type == 'pgdl' ~ 23
+	  ), n_prof = as.numeric(str_extract(exper_id, '[0-9]+')))
+	  
+	  # Save the processed data
+	  readr::write_csv(eval_data, file = file.path("2_process","out", processed_data_file))
```

Comment:
The `process_data()` function has two side effects: it creates the processed_data_file and the diagnostic_output_file. Can you modify to follow the "1:1 function-to-target pairing" recommended by the first bot comment on this PR?

Reply:
I'm split that function into 2 separate functions. Let me know if you have any other comments. (and no rush, I'm doing this as I have time in between other things)